src/util/heap/gc_trigger.rs

```diff
@@ -139,27 +147,203 @@ pub struct MemBalancerTrigger {
     max_heap_pages: usize,
     /// The current heap size
     current_heap_pages: AtomicUsize,
+    /// Statistics
+    stats: Atomic<MemBalancerStats>,
```

Comment:
The Atomic type is used on a complex struct.  The underlying implementation uses a global hash map to assign a spin lock for each hash value of address.  (See:   https://docs.rs/atomic/latest/src/atomic/fallback.rs.html#101 ) This is not really lock-free, and is not efficient, either.

Given that this struct is only used at GC start/release/end when no two threads access that struct concurrently, I suggest using AtomicRefCell and it’s borrow_mut method (you can implement access_stats with them).  It panics if two threads attempt to borrow it mutably at the same time, but given the use pattern, this shouldn’t happen unless there is a bug.

GC work packet stats are implemented using AtomicRefCell, too.

Alternatively you can always use Mutex.

Reply:
Done. I changed that to `AtomicRefCell`.